src/panels/lovelace/entity-rows/hui-media-player-entity-row.js

```diff
@@ -103,22 +103,40 @@ class HuiMediaPlayerEntityRow extends LocalizeMixin(PolymerElement) {
   _computeMediaTitle(stateObj) {
     if (!stateObj || this._isOff(stateObj.state)) return null;
 
+    // Default to guessing a prefix.
+    let prefix =
+      stateObj.attributes.media_title ||
+      stateObj.attributes.app_name ||
+      stateObj.state;
+
+    let suffix = "";
+
+    // If we know the media type then use a more specific prefix and suffix.
     switch (stateObj.attributes.media_content_type) {
       case "music":
-        return `${stateObj.attributes.media_artist}: ${
-          stateObj.attributes.media_title
-        }`;
+        prefix = stateObj.attributes.media_artist;
+        suffix = stateObj.attributes.media_title;
+        break;
       case "tvshow":
-        return `${stateObj.attributes.media_series_title}: ${
-          stateObj.attributes.media_title
-        }`;
-      default:
-        return (
-          stateObj.attributes.media_title ||
-          stateObj.attributes.app_name ||
-          stateObj.state
-        );
+        prefix = stateObj.attributes.media_series_title;
+        suffix = stateObj.attributes.media_title;
+        break;
     }
+
+    if (prefix && suffix) {
+      return `${prefix}: ${suffix}`;
+    }
+
```

Comment:
You initialize prefix/suffix as empty strings already so why not just add
```
else {
  return `${prefix}${suffix}`;
}
```

If one of them is defined, great; if not it will be an empty string as intended

Reply:
or one line it all
```
return `${prefix}${prefix && suffix ? ': ' : ''}${suffix}`;
```
🤷‍♂️

Reply:
The reason I can't just do `${prefix}${suffix}` is that in some cases `prefix` or `suffix` are re-defined as `undefined` by the `switch`;

```JavaScript
    switch (stateObj.attributes.media_content_type) {
      case "music":
        prefix = stateObj.attributes.media_artist;
        suffix = stateObj.attributes.media_title;
        break;
      case "tvshow":
        prefix = stateObj.attributes.media_series_title; // this is "undefined" on my Apple TV
        suffix = stateObj.attributes.media_title;
        break;
    }
```

Hence why I suggested the change here: https://github.com/home-assistant/home-assistant-polymer/pull/2060#issuecomment-439510389

Reply:
I think you can one line this logic kind of like Ian is suggesting. 

```suggestion
    return `${prefix || ''}${prefix && suffix ? ': ' : ''}${suffix || ''}`;
```

Reply:
That's similar to what I suggested in my comment above @zsarnett (but a bit more condensed) — if you think it's still clear in it's intent then I'll apply that change :+1:

Reply:
Go for it

Reply:
I've made that change :+1: